internal/provider/newrelic/provider_integration_test.go

```diff
@@ -123,6 +125,58 @@ func Test_Getting_external_metric_generates_a_query_not_rejected_by_backend(t *t
 	})
 }
 
+//nolint:paralleltest // This test registers environment variables, so it must not be run in parallel.
+func Test_Getting_external_metric_through_proxy(t *testing.T) {
+	ctx := testutil.ContextWithDeadline(t)
+
+	port := 1337
+	t.Setenv("HTTPS_PROXY", fmt.Sprintf("localhost:%d", port))
+
+	t.Run("ends_with_error_when_proxy_fails", func(t *testing.T) {
+		proxy := runProxy(t, port, func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
+			return goproxy.MitmConnect, host
```

Comment:
why using the `MitmConnect`, could you add a small comment?
If this "action" be used to mock different responses from the back end, it might worth to table driven this test and check for some different errors types

Reply:
My bad, It was intended to be `RejectConnect`.  The thing is I didn't find and easy way to mock different http responses and I end up with this approach of simulating a connection being rejected. However I could create a table driven test for these different `goproxy` called `Actions` like `RejectConnect, MitmConnect`. WDYT?

Reply:
> ohug! yes it should be that , what about using MainTest to orchestrate them WDYT?

AFAIK `MainTest` let you orchestrate the tests by adding some generic before and after behaviours in a per package way, but doesn't allow you to access each individual test per se; so the problem would be the same I think.

Reply:
I think just Reject would be fine for this test related to the Proxy. Te different status code response could go to the other test where you plan to check the provider resilience to errors i think.